test/testSelenium.js

```diff
@@ -163,18 +177,15 @@
         }
 
         if (!pushPayload) {
-          promise = webPush.sendNotification(subscription.endpoint, {
+          promise = webPush.sendNotification(subscription, null, {
             vapid: vapid
```

Comment:
Argh. This is exactly the kind of problem I'm trying to avoid :P 

I've added an extra check and test to avoid unexpected keys in the extra options.